sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarPartitionReaderWithPartitionValues.scala

```diff
@@ -16,102 +16,49 @@
 
 package com.nvidia.spark.rapids
 
-import ai.rapids.cudf.Scalar
-import com.nvidia.spark.rapids.Arm.{closeOnExcept, withResource}
-import com.nvidia.spark.rapids.RapidsPluginImplicits._
-
+import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.connector.read.PartitionReader
 import org.apache.spark.sql.execution.datasources.PartitionedFile
-import org.apache.spark.sql.types.{DataType, StructType}
+import org.apache.spark.sql.types.StructType
 import org.apache.spark.sql.vectorized.ColumnarBatch
 
+
 /**
  * A wrapper reader that always appends partition values to the ColumnarBatch produced by the input
  * reader `fileReader`. Each scalar value is splatted to a column with the same number of
  * rows as the batch returned by the reader.
  */
 class ColumnarPartitionReaderWithPartitionValues(
     fileReader: PartitionReader[ColumnarBatch],
-    partitionValues: Array[Scalar],
-    partValueTypes: Array[DataType]) extends PartitionReader[ColumnarBatch] {
+    partitionValues: InternalRow,
+    partitionSchema: StructType) extends PartitionReader[ColumnarBatch] {
   override def next(): Boolean = fileReader.next()
+  private var outputIter: Iterator[ColumnarBatch] = Iterator.empty
+
 
   override def get(): ColumnarBatch = {
-    if (partitionValues.isEmpty) {
+    if (partitionValues.numFields == 0) {
```

Comment:
Nit: Seems like this should be checking the schema not the row, since the schema dictates how many partition values there are and what types they have.  Comment applies to other places where this code was copied.